.github/workflows/docker-nightly-build.yaml

```diff
@@ -0,0 +1,52 @@
+name: docker-nightly-build
+
+on:
+  schedule:
+    # Set the time to be 20 mins after the pypi nightly build
+    - cron: '55 10 * * *' # 10:55am UTC, 2:55am PST, 5:55am EST
+  workflow_dispatch:
+  
+jobs:
+  
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up Python 3.10
+        uses: actions/setup-python@v4
+      - name: Find the release version
+        run: |
+          PACKAGE_NAME="skypilot-nightly"
+          # Fetch package info from PyPI
+          PACKAGE_INFO=$(curl -s https://pypi.org/pypi/$PACKAGE_NAME/json)
+          # Parse JSON and get the latest version
+          LATEST_VERSION=$(echo $PACKAGE_INFO | jq -r '.info.version')
+
+      - name: Setup Dockerfile
+        run: |
+          touch Dockerfile
+          cat <<EOF > Dockerfile
+          FROM continuumio/miniconda3:4.11.0
+
+          # Install dependencies
+          RUN conda install -c conda-forge google-cloud-sdk && \
+              apt update -y && \
+              apt install rsync vim -y && \
+              rm -rf /var/lib/apt/lists/*
+          
+          RUN pip install "skypilot-nightly[aws,gcp,azure,kubernetes]==$LATEST_VERSION" --no-cache-dir
```

Comment:
Might be worth a quick check to measure the size difference between an image with `skypilot-nightly[aws,gcp,azure,kubernetes]` vs `skypilot-nightly[all]`. If not too big, perhaps we can ship with `skypilot-nightly[all]`?

Reply:
Good call! I was using the older python version and the installation of [all] seems taking forever, due to some dependency issue with the kubernetes extra for python 3.9. I just switched the base image to use python 3.10, and it seems working nicely with `[all]`